### ISSUE TITLE
Confusing error message when missing parentheses in let-bound pattern match and tree structure

### ISSUE THREAD

**Issue by cartermp** [Microsoft/visualfsharp#6690]
Consider the following:

```fsharp
type T = Node of T *T
module T =
    let f t =
        let Node(l, r) = t
        l, r |> ignore
```

**Expected:**

This either:

* Works just fine
* Gives an error at the `let Node(l, r)` site

**Actual:**

This gives an error on `l` and `r` saying they're undefined:

<img width="800" alt="image" src="https://user-images.githubusercontent.com/6309070/57234812-56b68f00-6fd6-11e9-8e9f-bd939c99fc38.png">

The fix is to wrap the let-bound pattern match in parentheses:

```fsharp
type T = Node of T *T
module T =
    let f t =
        let (Node(l, r)) = t
        l, r |> ignore
```

**Comment by matthid** [comment]
> Gives an error at the let Node(l, r) site

Isn't it valid to define a function named 'Node' ?

**Comment by cartermp** [comment]
Right, didn't dawn on me. So in this case it's just a crappy error message

**Comment by cartermp** [comment]
Just for additional context I came across this in code like this:

if ...
else let (Node(l,r))= x in ...

And had to rewrite the pattern match in a small refactoring
